src/d3a/models/market/__init__.py

```diff
@@ -183,11 +47,11 @@ def lock_market_action(function):
     def wrapper(self, *args, **kwargs):
         # The market class needs to have an rlock member, that holds the recursive lock
         lock_object = getattr(self, RLOCK_MEMBER_NAME)
-        lock_object.acquire()
-        try:
+        if ConstSettings.GeneralSettings.EVENT_DISPATCHING_VIA_REDIS:
```

Comment:
Shouldn't this be `if not` ?

Reply:
Actually not. This lock is currently only used for the external connection, which is, for now, only allowed via EVENT_DISPATCHING_VIA_REDIS=False. The reason that we cannot have thread locks in this case, is because these locks do not allow multiple threads to run these functions concurrently, which is something that we need for the event dispatching via redis. I could rename this lock to something like 'external_connection_lock', if you think that this would make the code more understandable.

Reply:
Thanks! Yes, this could be helpful indeed.
 But I was thinking: If `ConstSettings.GeneralSettings.EVENT_DISPATCHING_VIA_REDIS` is set to False (default) this lock is acquired, so even for "normal" simulation runs. Doesnt this slow the simulation down?

Reply:
The performance impact of a lock operation is less than a microsecond according to [this benchmark](https://stackoverflow.com/a/30115176) . Therefore I do not think it is that critical to add. Also keep in mind that this is a reentrant lock, therefore the same thread is able to acquire the thread as many times as it wants without blocking. Therefore the single-threaded software run will never really block here.